### ISSUE TITLE
Auto save with Strip trailing spaces is really annoying 

### ISSUE THREAD

**Issue by van-de-bugger** [geany/geany#678]
Copied from [SourceForge](http://sourceforge.net/p/geany/bugs/835/):

When I have both auto-save and also strip trailing spaces turned on, I find a couple of problems if I am typing when auto-save triggers (i.e. quite often).

* If I am just about to type at the start of a new line when the auto save triggers then the current indent disappears and my cursor is left at the beginning of the line.
* If auto-save triggers while typing and I have just typed a space that space will disappear and my typing will continue, error included.

I can see this is consequence of my choice of preferences, but I still think there is a case for improvement here. The simple solution (without knowing the codebase at all) would seem to be to turn off the stripping temporarily for auto-saves. Is this possible?

And yes, I know it is 2012 but I have found myself in a situation where auto-save is a necessity :(

Geany 0.21.

Today it is 2015, Geany 1.25, everything is in place.

**Comment by elextr** [comment]
I'm not sure I understand what you intend, are you bumping the sourceforge entry on autosave stripping spaces?

**Comment by van-de-bugger** [comment]
Yes.

**Comment by elextr** [comment]
Ok

**Comment by codebrainz** [comment]
The semi-related PR #128 would probably solve this to some extent.

**Comment by mpetavy** [comment]
This is still an issue in 2026 with 2.1:
Auto-save and Strip trailing spaces removes typed trailing SPACE characters.
Will there be a fix for that?

**Comment by van-de-bugger** [comment]
@mpetavy: 

> Will there be a fix for that?

Yes and no. There is a fix, but I see no chance it will be merged upstream.

Background:

Four years ago I have developed _a few_ patches, including one to address this issue. I wanted to push all my patches upstream, but I started with one, the simplest patch, just to see how the submission process worked. I created a pull request, elextr asked for better documentation, I documented the feature, elextr reviewed the patch again and said "Looks ok by quick inspection".

That's all. A year later the patch still hadn't been merged, because one reviewer is not enough. Somebody else should review the patch, but there were no volunteers.

Since I failed to push even my  simplest patch upstream, so now I'm maintaining a dozen of my own patches locally.

Look at the project's backlog: there are almost 1,000 issues. I'm not blaming the project maintainers (no one is obligated to fix these bugs and close issues), I'm simply stating a fact: the project has stalled.

**Comment by eht16** [comment]
I don't know about the "fix" @van-de-bugger is referring to, I didn't find any reference.

Just to clarify:
you want to have the preference "Strip trailing spaces and tabs" enabled to trim trailing whitespace on save but don't want it to work on auto saving?

You could disable the feature and trigger it manually via the Document menu.

Otherwise it might need to identify all places where a file is saved automatically, across quite a few plugins.

If someone has a good idea how to properly do this and provide a PR, let's see.

**Comment by van-de-bugger** [comment]
@eht16:

> I don't know about the "fix" @van-de-bugger is referring to, I didn't find any reference.

Geany with my patches is available in [my personal repository](https://copr.fedorainfracloud.org/coprs/vandebugger/everything/). The source rpm contains all the patches. See the patch `geany-2.1-save-keeps-caret.vdb.patch`.

> If someone has a good idea how to properly do this and provide a PR, let's see.

Thanks, but I don't want to waste my time preparing a PR.
